scss/_buttons.scss

```diff
@@ -77,219 +104,520 @@
   // OUDS mod: no `.btn-check:checked:focus-visible + &` to handle box shadow
 
   // OUDS mod: selectors are slightly different to handle `.btn-check`
+  // TODO how to handle to disable links with aria-disabled only
   .btn-check:disabled + &,
-  &:disabled,
+  &:disabled:not(.loading-indeterminate):not(.loading-determinate):not(.skeleton),
   &.disabled,
   fieldset:disabled & {
     color: var(--#{$prefix}btn-disabled-color);
-    pointer-events: none;
     background-color: var(--#{$prefix}btn-disabled-bg);
     background-image: if($enable-gradients, none, null);
     border-color: var(--#{$prefix}btn-disabled-border-color);
     opacity: var(--#{$prefix}btn-disabled-opacity);
     @include box-shadow(none);
   }
+
+  .btn-check:disabled + &,
+  &:disabled,
+  &.disabled,
+  fieldset:disabled & {
+    pointer-events: none;
+  }
   // End mod
 
-  // OUDS mod: with icon
-  &:not(.btn-icon) > svg {
-    transform: translateY(1px);
+  // OUDS mod: buttons with icon and text
+  &:not(.btn-icon):has(> svg),
+  &:not(.btn-icon):has(> img),
+  &:not(.btn-icon):has(> .icon) {
+    --#{$prefix}btn-padding-start: #{$ouds-button-space-padding-inline-icon-start};
+    --#{$prefix}btn-padding-end: #{$ouds-button-space-padding-inline-start-icon-end};
+  }
+
+  &:not(.btn-icon) > svg,
+  &:not(.btn-icon) > img,
+  &:not(.btn-icon) > .icon {
+    width: $ouds-button-size-icon;
+    height: $ouds-button-size-icon;
+    margin-right: $ouds-button-space-column-gap-icon;
+    overflow: hidden;
+    font-size: $ouds-button-size-icon;
+    line-height: $ouds-button-size-icon;
   }
   // End mod
-}
 
+  // OUDS mod: loading buttons
+  @keyframes rotate-determinate {
+    0% { stroke-dashoffset: 315 #{"/* rtl:ignore */"}; }
+    100% { stroke-dashoffset: 0 #{"/* rtl:ignore */"}; }
+  }
+
+  @keyframes rotate1-indeterminate {
+    0%,
+    10% { stroke-dashoffset: 560 #{"/* rtl:ignore */"}; }
+    50%,
+    60% { stroke-dashoffset: 280 #{"/* rtl:ignore */"}; }
+    100% { stroke-dashoffset: 0 #{"/* rtl:ignore */"}; }
+  }
```

Comment:
LM, you changed the values in here and I find it is not working well on my side... To be seen together

Reply:
Changed it, it should be better now.